rfc-142-add-interest-cohort-permssion-policy-header.md

```diff
@@ -0,0 +1,52 @@
+
+## Problem
+
+In the short term the addition of FLoC actually increases the amount of information available to external observers, as FLoC can be used in conjunction with 3rd party cookies, as well as traditional browser fingerprinting techniques. When these techniques are used together, an observer's ability to identify an individual user increases. FLoC also circumvents some mitigations already put in place to limit data shared across domains. Browsers are rolling out partitioned storage, which restricts observers to only see information from their single domain. As the FLoC ID is generated from the browser history from a range of sites, FLoC actually results in the leaking of more data. This results in the observer seeing more information than they would without FLoC enabled.
+
+The FLoC specification specifically mentions that it “will never be able to prevent all misuse”, and the GOV.UK domain contains guidance and links to subjects that would be considered very sensitive for users, e.g.:
+
+- Universal credit login
+- Bankruptcy
+- Birth, marriage, deaths
+- Disability
+- Crime, justice and the law
+
+The FLoC specification does suggest that for sensitive cagerogies:
+
+> As a first mitigation, the browser should remove sensitive categories from its data collection. But this does not mean sensitive information can’t be leaked.
+
+However it does not guarantee sensitive information can't be leaked.
+
+To ensure information isn't leaked, an [explicit header can be set](https://github.com/WICG/floc#opting-out-of-computation).
+
+## Proposal
+
+To safeguard our users' sensitive browsing data, GOV.UK should explicitly set this header.
+
+We have 3 options for adding the header to the HTTP response: on the CDN, in the application, and in Nginx.
+
+- **Configure in CDN** - configure Fastly so it sends the header on each request.
+- **Configure in app** - configure the Rails apps to send the header.
+- **Configure in Nginx** - configure Nginx so it sends the header on each request.
+
+### Trade-offs
+
+| | Configure in CDN | Configure in app | Configure in Nginx |
+| --- | --- | --- | --- |
+| Deployment | The CDN is relatively easy and fast to deploy | Slow to roll out and iterate. We'd probably add it to `govuk_app_config`, which requires a version bump in multiple applications. Allows staged rollout. | Slow deployments via Puppet |
+| Policies | The header is set consistently for all of the requests, even ones that aren't served from a Rails app like [Licensing](https://github.com/alphagov/licensify) | Allows us to target per app and potentially per document type | Allows sharing between the frontend apps and publisher apps |
+
+### Preferred option
+
+We'll configure the following header in the CDN to ensure consistency across GOV.UK
```

Comment:
Thanks for opening this as per our Slack chat this is just a quick comment, with a couple of food for thought things. 

Might want to call out that this excludes all the internal apps as they aren't behind the CDN. I'd imagine we'd also want them excluded, which may make Nginx an option that provides a higher chance of platform consistency (this would exclude www.gov.uk/alerts though but that is already setting it's own permissions policy).

Also the www.gov.uk/alerts usage gave me a thought, would we append to an existing permissions-policy an app responds with or overwrite it? Rails 6.1 has introduced a [framework for setting them](https://github.com/alphagov/content-publisher/blob/d9de0e3236ed4172a9715f34c13ab37d65eb5ff5/config/initializers/permissions_policy.rb) that none of our apps have embraced yet (to my knowledge) so may be a point to decide whether apps can/should set a permissions policy

Reply:
Yeah that's a good point about publishing apps Kevin. Updated.

WRT Rails' framework for setting permission-policy, I agree it would be good to have a rule of thumb for GOV.UK, but I wonder whether this PR is the right place to set that guidance? Interested in your thoughts.

Reply:
Sure. Yes I don't think much of the `Permission-Policy` header stuff belongs in this PR. The area I think should be covered by this proposal is what Nginx (or CDN if that's decided) should do if a proxied app returns the header.  Options seem to be: assume app knows better do nothing; append to app's header to merge or overwrite the header. All seem to have pro's and con's, I'd maybe lean towards the first option myself for lowest deviation between dev and prod.

Reply:
I'd be in favour of the simplest thing we can get away with. If we just  think we should just rely on `add_header` at Nginx without imposing any more logic at the moment. As I understand it, that would essentially append to an app's header values.

I think this would give us a reasonable starting point. I suspect if we make use of the Permission-Policy header at an app level it will be for things other than FLoC. In those cases, we at least wouldn't run the risk of removing the explicit opt-out. If we do bump into a situation where we want to enable FLoC for specific apps, we can worry about the logic then.